pkg/controller/pods.go

```diff
@@ -94,14 +95,50 @@ func (cont *AciController) handlePodUpdate(pod *v1.Pod) bool {
 	}
 
 	cont.indexMutex.Lock()
-	defer cont.indexMutex.Unlock()
-
 	if pod.Spec.NodeName != "" {
 		// note here we're assuming pods do not change nodes
 		cont.addPodToNode(pod.Spec.NodeName, podkey)
 
 	}
-
+	nmport := false
+	for _, ctr := range pod.Spec.Containers {
+		for _, ctrportspec := range ctr.Ports {
+			if ctrportspec.Name != "" {
+				key := portProto(&ctrportspec.Protocol) + "-" + strconv.Itoa(int(ctrportspec.ContainerPort))
+				portmap, ok := cont.ctrPortNameMap[ctrportspec.Name]
+				if !ok {
+					cont.ctrPortNameMap[ctrportspec.Name] = make(map[string]uint)
+					cont.ctrPortNameMap[ctrportspec.Name][key]++
+					nmport = true
+				} else {
+					if _, found := portmap[key]; !found {
+						nmport = true
+					}
+					portmap[key]++
```

Comment:
I think this refcount won't be reliable. You might need to keep a reverse map.

Reply:
Ok will maintain the reverse map